pypika/dialects.py

```diff
@@ -328,33 +340,40 @@ def do_update(self, update_field, update_value):
             field = update_field
         self._on_conflict_updates.append((field, ValueWrapper(update_value)))
 
+    def _distinct_sql(self, **kwargs):
+        if self._distinct_on:
+            return "DISTINCT ON({distinct_on}) ".format(
+                distinct_on=",".join(
+                    term.get_sql(with_alias=True, **kwargs) for term in self._distinct_on
+                )
+            )
+        return super()._distinct_sql(**kwargs)
+
     def _conflict_field_str(self, term):
         if self._insert_table:
             return Field(term, table=self._insert_table)
 
     def _on_conflict_sql(self, **kwargs):
         if not self._on_conflict_do_nothing and len(self._on_conflict_updates) == 0:
-            if not self._on_conflict_field:
+            if not self._on_conflict_fields:
```

Comment:
the logic in this function is really complex. Looks like it was already complex even before you change, but would you mind simplifying it a bit to make it easier to read?

There are a couple of places where the indentation is redundant. For example, this if branch returns, so the else statement is redundant. The second else is also redundant because every branch of the if statement returns or raises an exception. I think getting rid of these redundant branches will make the code easier to follow because there will be less branching.